### PR TITLE
test: enables local tracing in e2e benchmark tests

### PR DESCRIPTION
**Pull request by staheri14**
Closes https://github.com/celestiaorg/celestia-app/issues/3318
It enables 1) local tracing in validators knuu instances 2) pushing locally traced data from validators to a S3 bucket, given that proper configs are passed via environment vars.

Additionally, it includes changes that:
- Allow specifying the number of blobs per sequence parameter for TX clients (previously fixed at `5`).
- Log information about the presence of Grafana environment variables.

## How to run
The `E2EThroughput` has been tested and I confirm both local tracing and pushing traces work correctly in knuu instances. To run the test:
```
go run ./test/e2e/benchmark  -v
```
You will see the `consensus_round_state.jsonl` file added to the root directory of the project, indicating the local tracing is operational in the validators' knuu instances.

In order to push traces to a S3 bucket, you would need to  pass S3 bucket push configs as env variables. I also tested this feature, and it works fine. It downloads the traced data and places them under `./traces/` folder in the project root.

### CODE REVIEW

File: test/e2e/benchmark/throughput.go [hunk 1]
```diff
@@ -49,6 +50,9 @@ func E2EThroughput() error {
 		Prometheus:         true,
 		GovMaxSquareSize:   appconsts.DefaultGovMaxSquareSize,
 		MaxBlockBytes:      appconsts.DefaultMaxBytes,
+		LocalTracingType:   "local",
+		PushTrace:          false,
```
Comment (staheri14) [inline test/e2e/benchmark/throughput.go:54]: [To reviewers] If you wish to push traced data, set these to true.

File: test/e2e/benchmark/benchmark.go [hunk 2]
```diff
@@ -63,7 +65,33 @@ func (b *BenchmarkTest) SetupNodes() error {
...
+		testnet.WithLocalTracing(b.manifest.LocalTracingType),
 	))
+
+	if b.manifest.PushTrace {
+		log.Println("reading trace push config")
+		if pushConfig, err := trace.GetPushConfigFromEnv(); err == nil {
+			log.Print("Setting up trace push config")
+			for _, node := range b.Nodes() {
+				if err = node.Instance.SetEnvironmentVariable(trace.
+					PushBucketName, pushConfig.BucketName); err != nil {
```
Comment (rootulp) [inline test/e2e/benchmark/benchmark.go:77]: uber nit, it seems odd to introduce a newline here

```suggestion
				if err = node.Instance.SetEnvironmentVariable(trace.PushBucketName, pushConfig.BucketName); err != nil {
```

File: test/e2e/benchmark/manifest.go [hunk 3]
```diff
@@ -68,6 +68,14 @@ type Manifest struct {
 	// consensus parameters
 	MaxBlockBytes int64
 
+	// other configs
```
Comment (rootulp) [inline test/e2e/benchmark/manifest.go:71]: is this line necessary considering there's another `//other configs` slightly below?

```suggestion
```

File: test/e2e/benchmark/manifest.go [hunk 4]
```diff
@@ -68,6 +68,14 @@ type Manifest struct {
...
+	// tracing configs, can be local or noop
```
Comment (rootulp) [inline test/e2e/benchmark/manifest.go:72]: perhaps clearer

```suggestion
	// LocalTracingType can be "local" or "noop".
```

File: test/e2e/testnet/node.go [hunk 5]
```diff
@@ -41,8 +44,24 @@ type Node struct {
...
 
-	rpcProxyPort  int
-	grpcProxyPort int
+	rpcProxyPort   int
+	grpcProxyPort  int
+	traceProxyPort int
+}
+
+// PullRoundStateTraces retrieves the round state traces from a node.
+// it will save it to the provided path.
```
Comment (rootulp) [inline test/e2e/testnet/node.go:53]:
```suggestion
// It will save them to the provided path.
```

**Review by rootulp — DISMISSED** [review]

File: test/e2e/testnet/node.go (hunk 5 above)
Comment (staheri14) [reply test/e2e/testnet/node.go]: Addressed https://github.com/celestiaorg/celestia-app/pull/3514/commits/2df5bf382a3660f3ffe8c30a63d4aafabc4ba2f6

File: test/e2e/benchmark/manifest.go (hunk 4 above)
Comment (staheri14) [reply test/e2e/benchmark/manifest.go]: Addressed https://github.com/celestiaorg/celestia-app/pull/3514/commits/2df5bf382a3660f3ffe8c30a63d4aafabc4ba2f6

File: test/e2e/benchmark/benchmark.go (hunk 2 above)
Comment (staheri14) [reply test/e2e/benchmark/benchmark.go]: Agree, fixed https://github.com/celestiaorg/celestia-app/pull/3514/commits/2df5bf382a3660f3ffe8c30a63d4aafabc4ba2f6

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — DISMISSED** [review]

**Review by rach-id — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**